sway/config/output.c

```diff
@@ -351,6 +356,22 @@ static int compute_default_scale(struct wlr_output *output) {
 	return 2;
 }
 
+/* Lists of formats to try, in order, when a specific render bit depth has
+ * been asked for. The last format in the list should always be XRGB8888,
+ * as a reliable backup in case the others are not available. */
```

Comment:
I'd prefer to not fallback to 8-bit when the user explicitly asks for 10-bit. Similar to how we handle modes which cannot be enabled.

Reply:
> Similar to how we handle modes which cannot be enabled.

It looks like sway does different things here, depending on the backend.
* With the DRM backend, it picks the closest listed mode to the one asked for; `https://github.com/swaywm/sway/blob/master/sway/config/output.c#L257-L275`
* Setting a --custom mode, or a modeline, tries to set the specified mode (and possibly freezes the screen if the mode isn't allowed.)
* With the Wayland backend, it always accepts the mode.

>  not fallback to 8-bit when the user explicitly asks for 10-bit.

If I pick this behavior, then Sway will crash if no 10 bit buffer formats are supported, because it does not have any buffers to render onto. This _is_ an experimental command, so it might be fine to do this as long as the documentation warns about it.

Reply:
> It looks like sway does different things here, depending on the backend.

Hm that's a good point. I forgot the bits about the DRM backend. We should look at a way to unify all of this. Since black screens aren't great, maybe we should always fallback to something that works, and integrate with swaynag to print warnings "your output mode couldn't be enabled" and "10-bit not available for this output".

Ideally we should also add a `render_bit_depth` field in `swaymsg -t get_outputs`, to let users query the current bit depth. But this can come later.

Also there's the wlr-output-management protocol, which never expects such fallbacks. It expects the configuration to fail if it's not possible. The wlr-output-management client itself will then try something else.

> If I pick this behavior, then Sway will crash if no 10 bit buffer formats are supported, because it does not have any buffers to render onto.

I wouldn't expect it to crash, but maybe it won't light up outputs configured with 10-bit render formats? But per the above, I've changed my mind.